app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java

```diff
@@ -0,0 +1,181 @@
+    private Resources r;
+
+    private int shortWait = 1000;
+    private int longWait = 4000;
+
+    private EditText emailText;
+    private EditText passwordText;
+
+    @Rule
+    public ActivityTestRule<LoginActivity> rule =
+            new ActivityTestRule<>(LoginActivity.class, true, true);
+
+    /**
+     * Runs before all tests and creates solo instance.
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception{
+        solo = new Solo(getInstrumentation(),rule.getActivity());
+
+        /* Gets context for app resources */
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+
+        /* Gets resource files */
+        r = appContext.getResources();
+
+        emailText = (EditText) solo.getView(R.id.login_email);
+        passwordText = (EditText) solo.getView(R.id.login_password);
+    }
+
+    /**
+     * Gets the activity
+     *
+     * @throws Exception
+     */
+    @Test
+    public void start() throws Exception {
+        Activity activity = rule.getActivity();
+    }
+
+    /**
+     * Tests going to CreateAccountActivity
+     */
+    @Test
+    public void createAccountButtonTest()
+    {
+        solo.clickOnButton(r.getString(R.string.create_account));
+        solo.assertCurrentActivity(r.getString(R.string.wrong_activity), CreateAccountActivity.class);
+
+        solo.sleep(shortWait);
+    }
+
+    /**
+     * Tests login button functionality with no input
+     */
+    @Test
+    public void emptyLoginTest()
+    {
+        solo.clickOnButton(r.getString(R.string.login));
+        solo.assertCurrentActivity(r.getString(R.string.wrong_activity), LoginActivity.class);
+
+        solo.sleep(shortWait);
+    }
+
+    /**
+     * Test for spaces input
+     */
+    @Test
+    public void spacesLoginTest()
+    {
+        solo.enterText(emailText, " ");
+        solo.enterText(passwordText, " ");
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(shortWait);
+
+        Assert.assertNotNull(emailText.getError());
+        Assert.assertNotNull(passwordText.getError());
+    }
+
+    /**
+     * Tests for incorrect password error
+     */
+    @Test
+    public void wrongPasswordTest()
+    {
+        solo.enterText(emailText, r.getString(R.string.test_account1_email));
+        solo.enterText(passwordText, r.getString(R.string.wrong_pass));
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(longWait);
+
+        Assert.assertNotNull(passwordText.getError());
+    }
+
+    /**
+     * Test for invalid or non-existing email input
+     */
+    @Test
+    public void invalidEmailTest()
+    {
+        solo.enterText(emailText, r.getString(R.string.wrong_email));
+        solo.enterText(passwordText, r.getString(R.string.wrong_pass));
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(longWait);
+
+        Assert.assertNotNull(emailText.getError());
```

Comment:
Does it make sense to assert the exact error here? What if emailText.getError() is returning an invalid error? For example "input can not be empty"?